plugins/Diagnostics/Diagnostic/Diagnostic.php

```diff
@@ -0,0 +1,22 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\Diagnostics\Diagnostic;
+
+/**
+ * Performs a diagnostic on the system or Piwik.
```

Comment:
I think 'diagnostic' isn't quite the right word here. For the class name I think it's ok, but diagnostic tools can do more than just one or more checks. For example, a command that lists the types & number of archives in an archive tables (ie, temporary, invalidated, for periods, etc.), number of duplicates if any, number of rows per archive, etc., could be considered a diagnostic tool, but it doesn't perform a simple check. Since these classes do just perform checks, I think it would be good to refer to them as 'diagnostic checks' in documentation.

Also if there are any requirements or conventions implementing classes should follow (like using translations?), I think they should be mentioned here. Maybe it's important to note that Diagnostics are loaded through DI and thus can use constructor based injection?

One last thing: would it be possible to include an example in the docs? I guess users could just look at the code for an example, but I'm not sure how many people would go to the code when looking on developer.piwik.org.